src/game_initialization/create_engine.cpp

```diff
@@ -741,7 +746,7 @@ void create_engine::init_all_levels()
 		}
 
 		if(type == "sp" || type.empty() || (type == "hybrid" && !mp)) {
-			campaign_ptr new_sp_campaign(new campaign(data));
+			std::shared_ptr<campaign> new_sp_campaign = std::make_shared<campaign>(data);
```

Comment:
Can use `auto` here.

I haven't re-read the discussion about what the editor should do yet - I'll probably add more comments when I've had to think about it, but that won't be in the next couple of days.